src/modal-checkout/analytics/ga4/checkout-attempt.js

```diff
@@ -0,0 +1,33 @@
+import { getEventPayload, getProductDetails, sendEvent } from './utils';
+
+/**
+ * Event fired when switching between steps of the multi-step checkout flow.
+ *
+ * @param {string} action Action name for the event: 'continue' or 'back'.
+ */
+
+export const manageCheckoutAttempt = () => {
+	if ( 'function' !== typeof window.gtag ) {
+		return;
+	}
+
+	const { action_type, amount, currency, product_id, product_type, recurrence, referer, variation_id } = getProductDetails( 'modal-checkout-product-details' );
```

Comment:
In general, when defining these variables from the output of `getProductDetails`, some will probably not be included in the output. So to guard against `undefined` values being sent, we can set defaults like this (using `variation_id` as an example):

```suggestion
	const { action_type, amount, currency, product_id, product_type, recurrence, referer, variation_id = '' } = getProductDetails( 'modal-checkout-product-details' );
```

Reply:
Ahhh good call! I've added this to this file, and the others. Thanks!